src/plugins/controls/duration/duration.js

```diff
@@ -70,66 +77,76 @@ export default pluginFactory({
+                //how many time elapsed from the last tick ?
 
-                    var context = testRunner.getTestContext();
+                var context = testRunner.getTestContext();
 
-                    //store by attempt
-                    var itemAttemptId = `${context.itemIdentifier}#${context.attempt}`;
+                //store by attempt
+                var itemAttemptId = `${context.itemIdentifier}#${context.attempt}`;
 
-                    durationStore.getItem(itemAttemptId).then(function(duration) {
+                currentUpdatePromise = queue.serie(() => durationStore.getItem(itemAttemptId)
+                    .then(function(duration) {
                         var elapsed = self.stopwatch.tick();
                         duration = _.isNumber(duration) ? duration : 0;
                         elapsed = _.isNumber(elapsed) && elapsed > 0 ? elapsed / 1000 : 0;
 
                         //store the last duration
-                        durationStore.setItem(itemAttemptId, duration + elapsed);
-                    });
-                },
+                        return durationStore.setItem(itemAttemptId, duration + elapsed);
+                    })
+                );
+
+                return currentUpdatePromise;
+            }
+
+            //one stopwatch to count the time
+            self.stopwatch = timerFactory({
+                autoStart: false,
+            });
+
+            //update the duration on a regular basis
+            self.polling = pollingFactory({
+                action: updateDuration,
                 interval: refresh,
                 autoStart: false
             });
 
             //change plugin state
             testRunner
-
                 .after('renderitem', function() {
                     self.enable();
                 })
                 .on('enableitem', function() {
                     self.enable();
                 })
-
+                .before('move skip exit timeout error disableitem', function() {
+                    updateDuration().then(() => self.disable());
+                })
                 .before('move skip exit timeout pause', function() {
                     var context = testRunner.getTestContext();
                     var itemAttemptId = `${context.itemIdentifier}#${context.attempt}`;
 
-                    return getItemDuration(itemAttemptId).then(function(duration) {
-                        var params = {
-                            itemDuration: 0
-                        };
-                        if (_.isNumber(duration) && duration > 0) {
-                            params.itemDuration = duration;
-                        }
-
-                        // the duration will be sent to the server with the next request,
-                        // usually submitItem() or callItemAction()
-                        testRunner.getProxy().addCallActionParams(params);
-                    });
+                    return currentUpdatePromise
```

Comment:
In case of rejection the events chain will be interrupted. It could be better to catch errors and log them directly.
@jsconan what do you think ?